File: typescript-models/src/ConfigurationLists/ConfigurationList.ts
export interface ConfigurationList {
    /**
     * MongoDB ObjectId
     */
    _id?:                      string;
    applicableDataSegmentIds?: string[];
    configurations?:           Configuration[];
    name?:                     string;
    readOnly?:                 boolean;
    [property: string]: any;
}

export interface Configuration {
    /**
     * MongoDB ObjectId
     */
    _id?:        string;
    descriptor?: string;
    details?:    Detail[];
    name?:       string;
    [property: string]: any;
}

export interface Detail {
    /**
     * MongoDB ObjectId
     */
    _id?:        string;
    descriptor?: string;
    identifier?: string;
    [property: string]: any;
}

File: csharp-models/Nimbus.DataModels/Nimbus.DataModels/Models/ConfigurationLists/ConfigurationList.cs
namespace Nimbus.DataModels.Models.ConfigurationLists
{
    using System;
    using System.Collections.Generic;

    using System.Globalization;
    using MongoDB.Bson;
    using MongoDB.Bson.Serialization.Attributes;

    public partial class ConfigurationList
    {
        /// <summary>
        /// MongoDB ObjectId
        /// </summary>
        [BsonElement("_id")]
        public ObjectId Id { get; set; }

        [BsonElement("applicableDataSegmentIds")]
        public List<string> ApplicableDataSegmentIds { get; set; }

        [BsonElement("configurations")]
        public List<Configuration> Configurations { get; set; }

        [BsonElement("name")]
        public string Name { get; set; }

        [BsonElement("readOnly")]
        public bool? ReadOnly { get; set; }
    }

    public partial class Configuration
    {
        /// <summary>
        /// MongoDB ObjectId
        /// </summary>
        [BsonElement("_id")]
        public ObjectId Id { get; set; }

        [BsonElement("descriptor")]
        public string Descriptor { get; set; }

        [BsonElement("details")]
        public List<Detail> Details { get; set; }

        [BsonElement("name")]
        public string Name { get; set; }
    }

    public partial class Detail
    {
        /// <summary>
        /// MongoDB ObjectId
        /// </summary>
        [BsonElement("_id")]
        public ObjectId Id { get; set; }

        [BsonElement("descriptor")]
        public string Descriptor { get; set; }

        [BsonElement("identifier")]
        public string Identifier { get; set; }
    }
}
